clang/lib/Sema/TreeTransform.h

```diff
@@ -12128,6 +12128,12 @@ ExprResult TreeTransform<Derived>::TransformSourceLocExpr(SourceLocExpr *E) {
                                            getSema().CurContext);
 }
 
+template <typename Derived>
+ExprResult TreeTransform<Derived>::TransformPPEmbedExpr(PPEmbedExpr *E) {
+  // TODO: fully implement for tree transformations
+  return E;
+}
```

Comment:
Can an embed ever be dependent? I'm not sure I understand the todo

Reply:
This is probably worth additional test coverage -- I don't think it can be dependent because we don't let the use specify the type for the embedded data. I think Jean Heyd may have been thinking of a future where we support specifying a different type for the elements, at which point it could theoretically be dependent. I think this code can be removed today though. CC @ErichKeane for additional opinions.

Reply:
I can't think of anything that could make an EmbedExpr be dependent in any way (in fact, your dependence calculation is to just set to 'none').  It could PERHAPS be in a template, but since everything is parsed/dealt with by the PP (and never looked up otherwise), nothing inside it could be dependent.

The function needs to still exist, but I think the 'todo' needs to change to 'Note: This can never be dependent on account of the way it is'.